janusgraph-core/src/main/java/org/janusgraph/graphdb/query/graph/JointIndexQuery.java

```diff
@@ -100,9 +100,14 @@ public String toString() {
 
     @Override
     public JointIndexQuery updateLimit(int newLimit) {
-        JointIndexQuery ji = new JointIndexQuery(Lists.newArrayList(queries));
-        ji.setLimit(newLimit);
-        return ji;
+        JointIndexQuery queryWithNewLimit;
```

Comment:
Consider if sth like this wouldn't be more readable:
```suggestion
                List<Subquery> subqueries = (queries.size() == 1) ? Lists.newArrayList(queries.get(0).updateLimit(newLimit)) :
            Lists.newArrayList(queries);
        JointIndexQuery queryWithNewLimit = new JointIndexQuery(subqueries);
        queryWithNewLimit.setLimit(newLimit);
        return queryWithNewLimit;
```
Btw I also wondering what happens if queries count changes after you set this limit. From the class construction it seems it is possible to add additional subquery after setting this limit, wouldn't it affect this process?

Reply:
Currently JointIndexQuery instantiated in one place where no additions queries added after limit update. But it is not clean, so may be move this type of heuristics into https://github.com/JanusGraph/janusgraph/blob/master/janusgraph-core/src/main/java/org/janusgraph/graphdb/query/graph/GraphCentricQueryBuilder.java#L355